### PR TITLE
[TH2-3835] Pass load settings via 'onStart' method

### PR DESCRIPTION
**Pull request by cordwelt**
* call 'onStart' before each step
 * add 'onStop' method indicating load-step end or forced load stop
 * ability to load dictionary via generator context

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -165,17 +165,18 @@ clean {
 }
 
 dependencies {
-    api platform('com.exactpro.th2:bom:3.0.0')
+    api platform('com.exactpro.th2:bom:3.2.0')
 
-    implementation 'com.exactpro.th2:common:3.18.1'
-    implementation 'com.exactpro.th2:grpc-woodpecker:0.0.2'
+    implementation 'com.exactpro.th2:common:3.39.0'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle]:
```suggestion
    implementation 'com.exactpro.th2:common:3.39.2'
```
To solve the problem `Common-j does not confirm filtered messages`

File: build.gradle [hunk 2]
```diff
@@ -165,17 +165,18 @@ clean {
...
+    implementation 'com.exactpro.th2:grpc-woodpecker:0.1.0-TH2-3835-+'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle]:
```suggestion
    implementation 'com.exactpro.th2:grpc-woodpecker:0.2.0'
```

File: build.gradle [hunk 3]
```diff
@@ -165,17 +165,18 @@ clean {
...
 
-    implementation 'org.slf4j:slf4j-log4j12'
-    implementation 'org.slf4j:slf4j-api'
+    implementation 'org.apache.logging.log4j:log4j-slf4j-impl'
+    implementation 'org.apache.logging.log4j:log4j-1.2-api'
+    implementation 'org.apache.logging.log4j:log4j-api'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle]: We should use slf4j only in projects to provide the ability to switch between implementation in common

File: src/main/kotlin/com/exactpro/th2/woodpecker/Main.kt [hunk 4]
```diff
@@ -122,11 +134,13 @@ fun main(args: Array<String>) = try {
 }
 
 data class Settings(
+    val minBatchesPerSecond: Int = 1,
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/woodpecker/Main.kt]: Please add the option description to README file

Comment (Nikita-Smirnov-Exactpro) [reply src/main/kotlin/com/exactpro/th2/woodpecker/Main.kt]: And for other options

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/exactpro/th2/woodpecker/Service.kt [hunk 5]
```diff
@@ -113,9 +118,11 @@ class Service(
             onInfo { "Started load cycle ${cycle + 1}" }
 
             forEach { step ->
+                onStart(step.settings.readSettings())
                 onInfo { "Started load step: ${step.toHuman()}" }
-                repeat(step.duration) { yield(step.rate) }
+                repeat(step.duration * minBatchesPerSecond) { yield(step.rate / minBatchesPerSecond) }
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/woodpecker/Service.kt]: Is it correct to divide the rate by minBatchesPerSecond?
rate - messages per second 
minBatchesPerSecond - batch per second

Comment (cordwelt) [reply src/main/kotlin/com/exactpro/th2/woodpecker/Service.kt]: Yep, the more often we generate batches the less their size must be to sum to requested load value

**Review by RaevskiySV — APPROVED** [review]

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**